applications/vanilla/controllers/class.discussioncontroller.php

```diff
@@ -158,6 +172,14 @@ public function index($DiscussionID = '', $DiscussionStub = '', $Page = '') {
         if (empty($canonicalUrl)) {
             $this->canonicalUrl(discussionUrl($this->Discussion, pageNumber($this->Offset, $Limit, 0, false)));
         } else {
+            $canonicalMessage = new Message(
+                sprintf(
+                    \Gdn::locale()->translate('This discussion has a more <a href="%1$s">recent version</a>.'),
```

Comment:
Any reason for the longer form here?

I think `\Gdn::translate()` is the standard.

Reply:
My original thought process:
1. I want to avoid using `Gdn` functions where possible.
1. `Gdn::locale` is a `Gdn` function, **but** it could be swapped out with a `Locale` instance if we are ever able to safely use DI with these old controllers.

I'm fine with swapping it out to `Gdn::translate`.